### ISSUE TITLE
Guarantee that we have price estimates for all tokens so that solver code is simpler

### ISSUE THREAD

**Issue by e00E** [gnosis/dex-services#1056]
https://github.com/gnosis/dex-services/issues/970#issuecomment-651025369
@marcovc 
> If the solver was given price estimates for all tokens as an input, then it wouldn't need to do any more fancy computations.

This should be guaranteed with the pricegraph and it would simplify the solver code significantly if I understand it correctly. We should verify this guarantee in our side and then get rid of the obsolete solver code.

**Comment by fleupold** [comment]
Yes, this is indeed the case for all tokens that are connected to the fee token. All other tokens cannot be included in the solution anyways. Do we have to filter out orders/balances for those tokens? Or does the solver ignore any orders for tokens that it doesn't have a price estimate for.

@marcovc

**Comment by marcovc** [comment]
Currently we estimate prices for all tokens that have no external price but are connected to the fee. We don't estimate prices for tokens not connected to the fee.

**Comment by fleupold** [comment]
This is blocked on getting more reliable estimates from the internal estimation (#1148)
